crates/red_knot_python_semantic/resources/mdtest/import/star.md

```diff
@@ -0,0 +1,418 @@
+# Wildcard (`*`) imports
+
+## Basic functionality
+
+### A simple `*` import
+
+`a.py`:
+
+```py
+X: bool = True
+```
+
+`b.py`:
+
+```py
+# TODO: should not error
+from a import *  # error: [unresolved-import]
+
+# TODO: should not error, should be `bool`
+# error: [unresolved-reference]
+reveal_type(X)  # revealed: Unknown
+
+print(Y)  # error: [unresolved-reference]
+```
+
+### Overriding existing definition
+
+`a.py`:
+
+```py
+X: bool = True
+```
+
+`b.py`:
+
+```py
+X = 42
+reveal_type(X)  # revealed: Literal[42]
+
+# TODO: should not error
+from a import *  # error: [unresolved-import]
+
+# TODO: should reveal `bool`
+reveal_type(X)  # revealed: Literal[42]
+```
+
+### Overridden by later definition
+
+`a.py`:
+
+```py
+X: bool = True
+```
+
+`b.py`:
+
+```py
+# TODO: should not error
+from a import *  # error: [unresolved-import]
+
+# TODO: should not error, should reveal `bool`
+# error: [unresolved-reference]
+reveal_type(X)  # revealed: Unknown
+
+X = 42
+reveal_type(X)  # revealed: Literal[42]
+```
+
+### Reaching across many modules
+
+`a.py`:
+
+```py
+X: bool = True
+```
+
+`b.py`:
+
+```py
+# TODO: should not error
+from a import *  # error: [unresolved-import]
+```
+
+`c.py`:
+
+```py
+from b import *
```

Comment:
the reason why we do _not_ emit an error here is somewhat horrifying... we're currently inserting a global-scope symbol with the name `"*"` in the symbol table for `b.py`